### ISSUE TITLE
Style/HashExcept wrong suggestion with array variables

### ISSUE THREAD

**Issue by j-seixas** [rubocop/rubocop#10754]
When using reject with an array variable to exclude keys from hash, the suggestion is wrong.

--------

## Expected behavior

```ruby
some_var = [:foo, :baz]
{foo: 1, bar: 2, baz: 3}.reject { |k, _v| k.include?(some_var) }
```

Expected message:
```
Use `except(*some_var)` instead. (convention:Style/HashExcept)
```

## Actual behavior

```
Use `except(some_var)` instead. (convention:Style/HashExcept)
```

## Steps to reproduce the problem

```ruby
some_var = [:foo, :baz]
{foo: 1, bar: 2, baz: 3}.reject { |k, _v| k.include?(some_var) }
# {bar: 2}

{foo: 1, bar: 2, baz: 3}.except(some_var) 
# {foo: 1, bar: 2, baz: 3}

{foo: 1, bar: 2, baz: 3}.except(*some_var) 
# {bar: 2}
```

## RuboCop version

```
$ bundle exec rubocop -V
1.31.0 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.1.0 x86_64-darwin20)
  - rubocop-graphql 0.14.3
  - rubocop-performance 1.14.2
  - rubocop-rails 2.15.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.11.1
```

**Comment by j-seixas** [comment]
related PR #10699

**Comment by j-seixas** [comment]
@koic Also found that it triggers a false positive when using reject on an Array
e.g.
```ruby
some_array = [:foo, :bar, :baz]
some_var = [:foo, :baz]
some_array.reject do |k, _v|
  some_var.include?(k)
end
```
will say to use `except` although that method doesn't exist for arrays
